### PR TITLE
[TypeScript][Skill] Migration to TypeScript 4

### PR DESCRIPTION
**Pull request by FedericoBernal**
Fix 3652

### Purpose
*What is the context of this pull request? Why is it being done?*
A new version of TypeScript has been released (v4)

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

- Updated TypeScript version to `^4.0.5` 
- Updated `p-queue` to `^6.0.0`
- Updated `eslint` to `^6.1.0`
- Updated `@typescript-eslint/eslint-plugin` to the version `^4.0.1`
- Updated `@typescript-eslint/parser` to the version `^4.0.1`
- Renamed rule `@typescript-eslint/no-object-literal-type-assertion` to `@typescript-eslint/no-use-before-define`
- Fixed ESLint issues
- Replicated changes to the Skill Template

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
We manually tested the execution of the Skill with TypeScript 4 and the execution of the unit tests.
image

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
\-

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: templates/typescript/samples/sample-skill/src/bots/defaultActivityHandler.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class DefaultActivityHandler<T extends Dialog> extends ActivityHandler {
         await this.conversationState.saveChanges(turnContext, false);
         await this.userState.saveChanges(turnContext, false);
     }
-
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (matiasroldan6) [inline templates/typescript/samples/sample-skill/src/bots/defaultActivityHandler.ts:43]: why is this needed now?

File: templates/typescript/samples/sample-skill/src/extensions/turnContextEx.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import { TurnContext } from 'botbuilder';
 import { SkillValidation, ClaimsIdentity } from 'botframework-connector';
 
+// eslint-disable-next-line @typescript-eslint/no-namespace
```
Comment (matiasroldan6) [inline templates/typescript/samples/sample-skill/src/extensions/turnContextEx.ts:9]: why is this needed now?

File: templates/typescript/samples/sample-skill/src/index.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ const defaultAdapter: DefaultAdapter = new DefaultAdapter(
     localeTemplateManager,
     telemetryInitializerMiddleware,
     telemetryClient);
-
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (matiasroldan6) [inline templates/typescript/samples/sample-skill/src/index.ts:114]: same as above :)

File: templates/typescript/samples/sample-skill/.eslintrc.json [hunk 4]
```diff
@@ -25,9 +25,7 @@
         "@typescript-eslint/indent": ["error", 4],
         "@typescript-eslint/interface-name-prefix": 0,
         "@typescript-eslint/no-explicit-any": 0,
-        "@typescript-eslint/no-object-literal-type-assertion": ["error", {
-            "allowAsParameter": true
-         }],
+        "@typescript-eslint/consistent-type-assertions": 1,
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/.eslintrc.json:28]: Following parity with #472 
```suggestion
        "@typescript-eslint/consistent-type-assertions": ["error", { "assertionStyle": "as" }],
```

**Review by Batta32 — CHANGES_REQUESTED** [review]
@matiasroldan6 check my feedback, also don't we have any warning to fix here with the bump of the eslint version?

File: templates/typescript/samples/sample-skill/.eslintrc.json (hunk 4 above)
Comment (matiasroldan6) [reply templates/typescript/samples/sample-skill/.eslintrc.json:28]: I checked and no new eslint issues appeared with the version bump, but will doublecheck

**Comment by matiasroldan6** [comment]
@Batta32 I checked and no new eslint issues appeared with the version bump, but will doublecheck

**Review by Batta32 — APPROVED** [review]
LGTM

**Pull request closed without merging**